### PR TITLE
Improving CONFIG test

### PR DESCRIPTION
**Pull request by harshmahesheka**
# 🎉 New feature

## Summary
Improves tests for CONFIG mode in ign_find_package to give error while finding wrong package as discussed in https://github.com/ignitionrobotics/ign-cmake/pull/230#pullrequestreview-940140554

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: examples/use_config_ifp/CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,18 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/")
 set(ignition-find_config_DIR "${CMAKE_SOURCE_DIR}/")
 #ign_find_package(ignition-find_config) will search for Findignition-find_config.cmake before ignition-find_configConfig.cmake 
 ign_find_package(ignition-find_config)
+if(CONFIG_TEST MATCHES "The package was found using MODULE mode")
+    message(STATUS "The package was found using MODULE mode")
+else()  
+    message( SEND_ERROR "Couldn't find correct package in MODULE mode.\nign_find_package CONFIG test failed")
```
Comment (adlarkin) [inline examples/use_config_ifp/CMakeLists.txt:11]: nit: remove whitespace next to opening parenthesis:
```suggestion
    message(SEND_ERROR "Couldn't find correct package in MODULE mode.\nign_find_package CONFIG test failed")
```

File: examples/use_config_ifp/CMakeLists.txt [hunk 2]
```diff
@@ -5,8 +5,18 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/")
...
+endif()
 #After CONFIG arguemnt ign_find_package will skip MODULE mode and search for ignition-find_configConfig.cmake directly
 ign_find_package(ignition-find_config CONFIG)
+if(CONFIG_TEST MATCHES "The package was found using CONFIG mode")
+    message(STATUS "The package was found using CONFIG mode")
+else()  
+    message( SEND_ERROR "Couldn't find correct package in CONFIG mode.\nign_find_package CONFIG test failed")
```
Comment (adlarkin) [inline examples/use_config_ifp/CMakeLists.txt:18]: nit: remove whitespace next to opening parenthesis:
```suggestion
    message(SEND_ERROR "Couldn't find correct package in CONFIG mode.\nign_find_package CONFIG test failed")
```

**Review by adlarkin — COMMENTED** [review]
Thanks for updating the test, this looks pretty good! I left a few minor comments below about removing extra whitespace.

Also, as per the [style guide](https://google.github.io/styleguide/cppguide.html#Spaces_vs._Tabs), our indents are 2 spaces, but it looks like you're using 4 spaces for indentation in the `CMakeLists.txt`. Would you mind fixing that as well?

**Comment by harshmahesheka** [comment]
Sorry for these mistakes, I will keep these in mind from the next time. Also, I have committed the required changes.

**Review by adlarkin — APPROVED** [review]
Thanks!

**Pull request merged**